client/configuration_variable.go

```diff
@@ -39,22 +40,30 @@ func (c *ConfigurationVariableSchema) ResourceDataSliceStructValueWrite(values m
 	return nil
 }
 
+type ConfigurationVariableOverwrites struct {
+	Value       string `json:"value"`
+	Regex       string `json:"regex"`
+	IsRequired  bool   `json:"isRequired"`
+	IsSensitive bool   `json:"isSensitive"`
+}
+
 type ConfigurationVariable struct {
-	ScopeId        string                       `json:"scopeId,omitempty"`
-	Value          string                       `json:"value" tfschema:"-"`
-	OrganizationId string                       `json:"organizationId,omitempty"`
-	UserId         string                       `json:"userId,omitempty"`
-	IsSensitive    *bool                        `json:"isSensitive,omitempty"`
-	Scope          Scope                        `json:"scope,omitempty"`
-	Id             string                       `json:"id,omitempty"`
-	Name           string                       `json:"name"`
-	Description    string                       `json:"description,omitempty"`
-	Type           *ConfigurationVariableType   `json:"type,omitempty" tfschema:",omitempty"`
-	Schema         *ConfigurationVariableSchema `json:"schema,omitempty"`
-	ToDelete       *bool                        `json:"toDelete,omitempty"`
-	IsReadOnly     *bool                        `json:"isReadonly,omitempty"`
-	IsRequired     *bool                        `json:"isRequired,omitempty"`
-	Regex          string                       `json:"regex,omitempty"`
+	ScopeId        string                           `json:"scopeId,omitempty"`
+	Value          string                           `json:"value" tfschema:"-"`
+	OrganizationId string                           `json:"organizationId,omitempty"`
+	UserId         string                           `json:"userId,omitempty"`
+	IsSensitive    *bool                            `json:"isSensitive,omitempty"`
+	Scope          Scope                            `json:"scope,omitempty"`
+	Id             string                           `json:"id,omitempty"`
+	Name           string                           `json:"name"`
+	Description    string                           `json:"description,omitempty"`
+	Type           *ConfigurationVariableType       `json:"type,omitempty" tfschema:",omitempty"`
+	Schema         *ConfigurationVariableSchema     `json:"schema,omitempty"`
+	ToDelete       *bool                            `json:"toDelete,omitempty"`
+	IsReadOnly     *bool                            `json:"isReadonly,omitempty"`
+	IsRequired     *bool                            `json:"isRequired,omitempty"`
+	Regex          string                           `json:"regex,omitempty"`
+	Overwrites     *ConfigurationVariableOverwrites `json:"overwrites,omitempty"` // Is removed when marhseling to a JSON.
```

Comment:
This struct is also sent in a POST requests.
Overwrites field should **not** be sent in a request. (We are only intrested of this value in a response).
To solve this issue without making too many code changes, I have decided to write a custom marshal.